keps/sig-node/20200203-memory-manager.md

```diff
@@ -0,0 +1,584 @@
+	Allocatable    uint64 `json:"allocatable"`
+	Reserved       uint64 `json:"reserved"`
+	Free           uint64 `json:"free"`
+}
+
+// NodeState contains NUMA node related information
+type NodeState struct {
+	// NumberOfAssignments contains a number memory assignments from this node
+	// When the container requires memory and hugepages it will increase number of assignments by two
+	NumberOfAssignments int `json:"numberOfAssignments"`
+	// MemoryTable contains NUMA node memory related information
+	MemoryMap map[v1.ResourceName]*MemoryTable `json:"memoryMap"`
+	// NodeGroups contains NUMA nodes that current NUMA node in group with them
+	// It means that we have container that pinned to the current NUMA node and all group nodes
+	Nodes []int `json:"nodes"`
+}
+
+// NodeMap contains memory information for each NUMA node.
+type NodeMap map[int]*NodeState
+```
+
+For instance, consider a two-socket machine with only one size of hugepages pre-allocated. In this case, *Memory Manager* instatiates four memory tables (`[ two NUMA nodes ] * [ two  memory types ]`) on start-up. Notice that on a single-socket machine, and in the case when sub-NUMA cluster (SNC) feature is enabled in hardware, *Memory Manager* can detect two NUMA nodes although, as already mentioned, the machine features a single socket.
+
+### Memory Map 
+
+`Memory Map` serves the purpose of tracking the memory utilization at each NUMA node. `Memory Map` involves several counters for tracking the memory utilization. The following rules hold:
+
+```go
+Allocatable = TotalMemSize - SystemReserved
+Free + Reserved = Allocatable
+```
+or:
+```go
+Free = Allocatable - Reserved 
+Reserved = Allocatable - Free
+```
+
+* The value of `TotalMemSize` is provided by `cadvisor` to *Memory Manager* for each memory type (regular memory, `hugepages-1Gi`, and so forth). The value of `TotalMemSize` is constant, and indicates the total (maximum) capacity of memory of a particular type available at a NUMA node. 
+* `SystemReserved` is equal to `0` (zero) unless the administrator of _K8S_ node decides to employ [Node Allocatable Feature](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/) which was designed for improving the stability of _K8S_ node. We elaborate below the characteristics of this feature. Also, we describe the way how *Memory Manager* must be configured to satisfy and remain backwards-compatible with [Node Allocatable Feature](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/). The value of `SystemReserved` depends on the pre-configuration passed to *Memory Manager* by the administrator of _K8S_ node through `--reserved-memory` flag. Details concerning the flag are present in [this](#reserved-memory-flag) section.
+* `Reserved` memory denotes exactly the same what we already meant by another term, i.e. **guaranteed** memory. So, in other words, `SystemReserved` stands for the memory **guaranteed** for system or `kubelet` daemons.
```

Comment:
I think we should re-phrase the `Reserved` section because it hard to understand that it contains the memory that already in use by running containers.
I think you should move the `Reserved` section that appears below to here - https://github.com/kubernetes/enhancements/pull/1203/files#diff-9b2b9d1d51c55bd057576a556900a693R262

Reply:
Thanks, Done. 
Rephrased to 
```
Reserved` indicates the total **guaranteed** memory reserved for containers in Guaranteed QoS class
```